drivers/wifi/wfx200/wfx200.c

```diff
@@ -0,0 +1,499 @@
+	case WFX200_STATE_AP_MODE:
+		if (!(context->sl_context.state & SL_WFX_AP_INTERFACE_UP)) {
+			status = -EIO;
+			goto out;
+		}
+		interface = SL_WFX_SOFTAP_INTERFACE;
+		break;
+	case WFX200_STATE_STA_MODE:
+		if (!(context->sl_context.state & SL_WFX_STA_INTERFACE_CONNECTED)) {
+			status = -EIO;
+			goto out;
+		}
+		interface = SL_WFX_STA_INTERFACE;
+		break;
+	default:
+		status = -ENODEV;
+		goto out;
+	}
+	frame_len = SL_WFX_ROUND_UP(len, 2);
+
+	result = sl_wfx_allocate_command_buffer((sl_wfx_generic_message_t **)&tx_buffer,
+						SL_WFX_SEND_FRAME_REQ_ID,
+						SL_WFX_TX_FRAME_BUFFER,
+						frame_len + sizeof(sl_wfx_send_frame_req_t));
+	if (result != SL_STATUS_OK) {
+		status = -ENOMEM;
+		goto out;
+	}
+
+	if (net_pkt_read(pkt, tx_buffer->body.packet_data, len)) {
+		status = -EIO;
+		goto out_free;
+	}
+
+	result = sl_wfx_send_ethernet_frame(tx_buffer,
+					    frame_len,
+					    interface,
+					    WFM_PRIORITY_BE0);
+	if (result != SL_STATUS_OK) {
+		status = -EIO;
+		goto out_free;
+	}
+
+out_free:
+	sl_wfx_free_command_buffer((sl_wfx_generic_message_t *)tx_buffer,
+				   SL_WFX_SEND_FRAME_REQ_ID,
+				   SL_WFX_TX_FRAME_BUFFER);
+out:
+	return status;
+}
+
+static enum ethernet_hw_caps wfx200_get_capabilities(const struct device *dev)
+{
+	ARG_UNUSED(dev);
+
+	return 0;
+}
+
+int wfx200_scan(const struct device *dev, scan_result_cb_t cb)
+{
+	struct wfx200_data *context = dev->data;
+	sl_status_t result;
+
+	if (context->state < WFX200_STATE_INTERFACE_INITIALIZED) {
+		return -EIO;
+	}
+	if (context->scan_cb != NULL) {
+		return -EINPROGRESS;
+	}
+	context->scan_cb = cb;
+	result = sl_wfx_send_scan_command(WFM_SCAN_MODE_PASSIVE,
+					  NULL,
+					  0,
+					  NULL,
+					  0,
+					  NULL,
+					  0,
+					  NULL);
+	if (result != SL_STATUS_OK) {
+		context->scan_cb = NULL;
+		return -EIO;
+	}
+	return 0;
+}
+
+int wfx200_connect(const struct device *dev, struct wifi_connect_req_params *params)
+{
+	struct wfx200_data *context = dev->data;
+	char ssid[WIFI_SSID_MAX_LEN + 1];
+	sl_wfx_security_mode_t mode;
+	sl_status_t result;
+	uint16_t channel;
+
+	switch (context->state) {
+	case WFX200_STATE_INTERFACE_INITIALIZED:
+		break;
+	case WFX200_STATE_AP_MODE:
+		return -EBUSY;
+	case WFX200_STATE_STA_MODE:
+		return -EALREADY;
+	default:
+		return -ENODEV;
+	}
+	memcpy(ssid, params->ssid, MIN(params->ssid_length, sizeof(ssid)));
+	ssid[MIN(params->ssid_length, WIFI_SSID_MAX_LEN)] = 0;
+
+	if (params->security == WIFI_SECURITY_TYPE_PSK) {
+		mode = WFM_SECURITY_MODE_WPA2_WPA1_PSK;
+	} else if (params->security == WIFI_SECURITY_TYPE_NONE) {
+		mode = WFM_SECURITY_MODE_OPEN;
+	} else {
+		LOG_ERR("Not supported security mode");
+		return -ENOTSUP;
+	}
+
+	if (params->channel <= WIFI_CHANNEL_MAX) {
+		channel = params->channel;
+	} else {
+		channel = 0;
+	}
+
+	LOG_INF("Connecting to %s on channel %d", log_strdup(ssid), channel);
+
+	result = sl_wfx_send_join_command(params->ssid,
+					  params->ssid_length,
+					  NULL,
```

Comment:
It would be nice if we could pass third parameter as well, [BSSID ](https://docs.silabs.com/wifi/wf200/rtos/latest/group-f-u-l-l-m-a-c-d-r-i-v-e-r-a-p-i#ga2fd76ed31e48be10ab6b7fb9d4bc454d)(i.e. MAC of the AP to which we are connecting).